### DISCUSSION TITLE
How to get asset details on catalog request

### DISCUSSION THREAD

**Discussion by SebastianOpriel** [eclipse-edc/Connector#3363] [Q&A] [answered]
We are currently struggeling with changes to the catalog endpoint responses. Previously we were able to get all asset information from a `ContractOffer` via [`getAssets()`](https://github.com/eclipse-edc/Connector/blob/c1dbbabf65157e95af8f9509b4e7639948291612/spi/control-plane/contract-spi/src/main/java/org/eclipse/edc/connector/contract/spi/types/offer/ContractOffer.java#L108) method, now only containing an assetId with method [`getAssetId()`](https://github.com/eclipse-edc/Connector/commit/110ef7744fda3910031430fd87007863b30a0c64#diff-e56f1393aca00a7168ec6291108eafe5c9f111d49e6f947966b6d657b843c5e4). 

In a catalog browser on consumer side, we want to visualize all asset offerings incl. details. The current idea would be to get the catalog of a provider and fetch each's asset information with a separate call to [getAsset endpoint](https://app.swaggerhub.com/apis/eclipse-edc-bot/management-api/0.2.0#/Asset/getAsset_1). I assume, this is not the intended way. Have you any recommendation?

**Comment by ndr-brt** [comment, marked as answer]
In a DCAT catalog, the `Dataset` structure (that refers to a single `Asset` in the EDC domain) should contain all the `Asset` properties., [please take a look at this Sample](https://github.com/eclipse-edc/Samples/tree/main/transfer/transfer-06-consumer-pull-http#6-how-to-fetch-catalog-on-consumer-side), in which the Catalog contains for example the `edc:contenttype` property, that was part of the asset created on the provider.

the `getAsset` endpoint is on the management side and it should not be accessed outside the owner organization.

**Reply by SebastianOpriel** [reply]
Thanks for your quick reply and the explanation!

**Reply by SebastianOpriel** [reply]
@ndr-brt One more question: Our EDC returns an array of datasets, 
```json
{
  "@id": "478a07bb-4df0-471e-aabe-c6d47558b329",
  "@type": "dcat:Catalog",
  "dcat:dataset": [
    {
    "@id": "test-1.0",
    "@type": "dcat:Dataset",
    ...
```
while the linked document returns an object
```json
{
  "@id": "31f6d748-d35b-4dec-9e34-d141fd17b458",
  "@type": "dcat:Catalog",
  "dcat:dataset": {
    "@id": "assetId",
    "@type": "dcat:Dataset",
    ...
```

Is the linked document correct? Thus, we shall expect for parsing the DCAT an object, if none or just one dataset is available, and otherwise an array?

**Reply by SebastianOpriel** [reply]
While it is here also an array: https://github.com/International-Data-Spaces-Association/ids-specification/blob/2546b7806bc7c69e23b24944ea75ebdc3e44d197/catalog/message/catalog.json#L19C8-L19C20

**Reply by ndr-brt** [reply]
This is because how the json-ld compaction is configured: ref. https://github.com/eclipse-edc/Connector/issues/3323

**Reply by SebastianOpriel** [reply]
Oh thanks :-) I understand, that we always can expect an array (in the future).
